packages/code/src/main.rs

```diff
@@ -0,0 +1,144 @@
+            if let Value::String(val) = &$input["arguments"][$arg] {
+                val
+            } else {
+                panic!("No theme argument was provided");
+            }
+        }
+    }
+    match to.as_str() {
+        "html" => {
+            let input: Value = {
+                let mut buffer = String::new();
+                io::stdin().read_to_string(&mut buffer).unwrap();
+                serde_json::from_str(&buffer).unwrap()
+            };
+
+            let code = input["data"].as_str().unwrap();
+            let lang = get_arg!(input, "lang");
+            let font_size = get_arg!(input, "font_size");
+            let tab_size = get_arg!(input, "tab_size");
+            let theme = get_arg!(input, "theme");
+            let bg = get_arg!(input, "bg");
+
+
+            let (highlighted, default_bg) = get_highlighted(code, lang, theme);
+            let style = get_style(font_size, tab_size, bg, default_bg);
+
+            let html = format!(r#"<pre {style}>{highlighted}</pre>"#);
+            let json = json!({"name": "raw", "data": html}).to_string();
+
+            print!("[{json}]");
+        }
+        other => {
+            eprintln!("Cannot convert code to {other}");
+            return;
+        }
+    }
+}
+
+fn get_style(font_size: &String, tab_size: &String, bg: &String, default_bg: Option<Color>) -> String {
+    let hex;
+    if bg == "default" && default_bg.is_some() {
+        let c = default_bg.unwrap();
+        hex = format!("{:02x}{:02x}{:02x}", c.r, c.g, c.b);
+    } else {
+        hex = bg.clone();
+    }
+
+    let mut style = String::from("style=\"");
+    write!(style, "box_sizing: border_box; ").unwrap();
+    write!(style, "padding: 0.5rem; ").unwrap();
+    write!(style, "tab-size: {}; ", tab_size).unwrap();
+    write!(style, "font-size: {}px; ", font_size).unwrap();
+    write!(style, "background-color: #{}; ", hex).unwrap();
+    style.push('\"');
+    style
+}
+
+fn get_highlighted(code: &str, lang: &String, tm: &String) -> (String, Option<Color>) {
+    let ss = SyntaxSet::load_defaults_newlines();
+    let ts = ThemeSet::load_defaults();
+    let syntax = match ss.find_syntax_by_token(lang) {
+        Some(sr) => sr,
+        _ => ss.find_syntax_by_token("py").unwrap()
+    };
+    let theme = match tm.as_str() {
+        "ocean_dark" => &ts.themes["base16-ocean.dark"],
+        "ocean_light" => &ts.themes["base16-ocean.light"],
+        "mocha" => &ts.themes["base16-mocha.dark"],
+        "eighties" => &ts.themes["base16-eighties.dark"],
+        "github" => &ts.themes["InspiredGitHub"],
+        "solar_dark" => &ts.themes["Solarized (dark)"],
+        "solar_light" => &ts.themes["Solarized (light)"],
+        _ => &ts.themes["InspiredGitHub"]
+    };
+    let mut h = HighlightLines::new(syntax, theme);
+    let incl_bg = IncludeBackground::No;
+    let mut html: Vec<String> = vec![];
+
+    // avoiding lines() here because we want to include the final newline
```

Comment:
I am unsure what you mean by this, it seems to me that `lines()` does the same thing as you do, and I don't really know what you mean by including the final newline. When I tested both ways, it looks like I get the same result regardless if I do it this way or using `lines()`

Reply:
Oh well, I see that if you do it your way, you add a new `<br>` at the end. But I fail to see why that is needed since the result looks the same anyways, and you can always just append a `<br>` yourself... But maybe there is a good reason, please tell me, I am curious!

Even doing this:
```
[block_content]"
[code python]
def foo():
    1 + 1"
Text afterwards
```

which would make sure text comes immediately after the code block without an empty line in between, using `.lines()`, seems to work fine for me.

Reply:
It gives different results when using delimiters.
For example:
```
[code python]{

def foo():
    print("foo")

}
```
When using ``lines()`` the empty row at the end will not be included, while it is included in the current implementation.

Reply:
Oh okay. Clever solution! Keep it as it is then :)